### ISSUE TITLE
how to make make_data_hetero() while using flow_from_dirctory 

### ISSUE THREAD

**Issue by tamaraalshekhli** [piccolbo/multitask-learn#1]
Thank you for sharing your code, I'm using flow_from_dirctory to generate my own data, in this case how i can add the noise similar to make_data_hetero and feed the data to the image generator and 2mask generators?
another question related to the (mean_squared_error_hetero) my both tasks are semantic segmentation, is this loss will work for segmentation task?

**Comment by piccolbo** [comment]
Hi, as to the first question I think it's best directed to a keras user group or support group (that is, I have no idea). As to the appropriateness of this function for image segmentation, I think the main application is combining disparate task. The original paper where I found about this function combines segmentation, depth mapping and another visual task. So it seems you are not in that specific situation having to combine two similar tasks. That said, one of the two segmentation problems could be "harder" for your chosen model, meaning resulting in higher error after training. In that case, it may make sense to estimate two variances, one per task, hence a version of the loss implemented here and proposed in that paper could be a good choice.

**Comment by tamaraalshekhli** [comment]
thank you for your reply, can you please elaborate more (estimate two variances, one per task, hence a version of the loss implemented here and proposed in that paper could be a good choice) because i have the situation that you mentioned after training (one of the two segmentation problems could be "harder" for your chosen model, meaning resulting in higher error after training)and i dont know how to solve it , my two tasks are correlated with each other (road and centerline extraction) and i used binary cross entropy in keras
